python/ffsim/states/sample_slater.py

```diff
@@ -211,3 +403,96 @@ def _generate_probs(rdm: np.ndarray, sample: set[int], norb: int) -> np.ndarray:
         indices = list(sample | {orb})
         probs[i] = np.linalg.det(rdm[indices][:, indices]).real
     return probs / np.sum(probs)
+
+
+class _FastProjectionSampler:
```

Comment:
This is really great, but in ffsim we actually prefer a functional programming style when possible. Do you think it's sensible to refactor this to use helper functions instead of a helper class while retaining identical performance? I think it would look like this:
- Define a function that computes the normal vectors
- Define a sampling function that takes the normal vectors as input, together with other needed information like `norb` and `nelec`